WireAPI/Tests/WireAPITests/Mocks/HTTPClientMock.swift

```diff
@@ -97,7 +97,7 @@ final class HTTPClientMock: HTTPClient {
 
     convenience init(
         code: Int,
-        payload: Data
+        payload: Data?
```

Comment:
Why is this change necessary? We should avoid optionals wherever possible.

Reply:
I believe because we encountered the first time a response has no payload data.

Reply:
exactly, we need to handle use cases where API succeeds but no payload is sent back. this is the case for the pushSupportedProtocols API.